### PR TITLE
Dashboards: Add support for `widgets` field

### PR DESCRIPTION
**Pull request by conner**
- Use codegen / add tests
- Small refactor of virtual tag test

### CODE REVIEW

File: vantage/dashboard_resource.go [hunk 1]
```diff
@@ -27,127 +26,46 @@ func NewDashboardResource() resource.Resource {
...
-					stringplanmodifier.UseStateForUnknown(),
-				},
-			},
-			"saved_filter_tokens": schema.ListAttribute{
-				ElementType:         types.StringType,
-				MarkdownDescription: "Tokens of the saved filters used in the Dashboard.",
-				Optional:            true,
-			},
+	s := resource_dashboard.DashboardResourceSchema(ctx)
+	attrs := s.GetAttributes()
```
Comment (conner) [inline vantage/dashboard_resource.go:35]: 🤷 use the generated schema with some overrides

File: vantage/dashboard_resource.go [hunk 2]
```diff
@@ -27,127 +26,46 @@ func NewDashboardResource() resource.Resource {
...
+		Computed:            true,
+		Optional:            true,
+		MarkdownDescription: attrs["end_date"].GetMarkdownDescription(),
+	}
+	s.Attributes["title"] = schema.StringAttribute{
+		Computed:            true,
+		Optional:            true,
+		MarkdownDescription: attrs["title"].GetMarkdownDescription(),
+	}
+	s.Attributes["token"] = schema.StringAttribute{
```
Comment (whereandy) [inline vantage/dashboard_resource.go:46]: Oh wow, it never occurred to me to do it this way. I want to change out all the places where I did now.

File: vantage/dashboard_resource.go [hunk 3]
```diff
@@ -267,19 +142,17 @@ func (r DashboardResource) Update(ctx context.Context, req resource.UpdateReques
...
+		resp.Diagnostics.Append(diag...)
+		return
+	}
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
 func (r DashboardResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
-	var state *DashboardResourceModel
+	var state *resource_dashboard.DashboardModel
```
Comment (whereandy) [inline vantage/dashboard_resource.go:155]: nit: use *dashboardModel for consistency

**Review by whereandy — APPROVED** [review]

File: vantage/dashboard_resource.go (hunk 2 above)
Comment (conner) [reply vantage/dashboard_resource.go:46]: Same - An experiment. I try to try at least one thing every time I dip my toe in here :: sisyphus emoji :: for making contributions here easier

Comment (conner) [reply vantage/dashboard_resource.go:46]: next stop: some json annotated struct <=> tfsdk struct field conversion patterns

**Pull request merged**